Meadowlark-js/backends/meadowlark-opensearch-backend/test/integration/QueryOpensearch.test.ts

```diff
@@ -0,0 +1,305 @@
+        invalidResourceInfo.projectName = 'wrong-project';
+        const result = await queryDocuments(setupQueryRequest({}, {}, invalidResourceInfo), client);
+
+        expect(result.response).toEqual('QUERY_FAILURE_INVALID_QUERY');
+        expect(result.documents).toHaveLength(0);
+        expect(result.totalCount).toBeUndefined();
+      });
+    });
+
+    describe('when querying with wrong property', () => {
+      it('should return error message', async () => {
+        const result = await queryDocuments(setupQueryRequest({ lastSure: 'Last' }, {}), client);
+
+        expect(result.response).toEqual('QUERY_SUCCESS');
+        expect(result.totalCount).toEqual(0);
+        expect(result.documents).toHaveLength(0);
+      });
+    });
+
+    describe('when querying with non existent data', () => {
+      it('should return empty results', async () => {
+        const result = await queryDocuments(setupQueryRequest({ firstName: 'Last' }, {}), client);
+
+        expect(result.response).toEqual('QUERY_SUCCESS');
+        expect(result.totalCount).toEqual(0);
+        expect(result.documents).toHaveLength(0);
+      });
+    });
+
+    describe('when querying without parameters', () => {
+      it('should return all values', async () => {
+        const result = await queryDocuments(setupQueryRequest({}, {}), client);
+
+        expect(result.response).toEqual('QUERY_SUCCESS');
+        expect(result.totalCount).toEqual(2);
+        expect(result.documents[0]).toEqual(student1);
+        expect(result.documents[1]).toEqual(student2);
+      });
+    });
+
+    describe('when querying with valid parameters', () => {
+      it('should return value', async () => {
+        const result = await queryDocuments(setupQueryRequest({ firstName: student1.firstName }, {}), client);
+
+        expect(result.response).toEqual('QUERY_SUCCESS');
+        expect(result.totalCount).toEqual(1);
+        expect(result.documents[0]).toEqual(student1);
+      });
+    });
+
+    describe('when querying with limit', () => {
+      it('should return value', async () => {
+        const result = await queryDocuments(setupQueryRequest({}, { limit: 1 }), client);
+
+        expect(result.response).toEqual('QUERY_SUCCESS');
+        expect(result.totalCount).toEqual(2);
+        expect(result.documents).toHaveLength(1);
+        expect(result.documents[0]).toEqual(student1);
+      });
+    });
+
+    describe('when querying with limit and offset', () => {
+      it('should return value', async () => {
+        const result = await queryDocuments(setupQueryRequest({}, { limit: 1, offset: 1 }), client);
+
+        expect(result.response).toEqual('QUERY_SUCCESS');
+        expect(result.totalCount).toEqual(2);
+        expect(result.documents).toHaveLength(1);
+        expect(result.documents[0]).toEqual(student2);
+      });
+    });
+
+    describe('when querying with parameters and offset', () => {
+      it.skip('should return value', async () => {
```

Comment:
@stephenfuqua this scenario:
`GET students/?firstName=Test&Offset=1 -> returns an empty array, since there is only one student to match`
Is returning UNKNOWN_FAILURE instead of the mentioned result.
Is this the correct behavior?

Reply:
Agreed, this is an invalid scenario with OpenSearch, and I simply forgot about that. Can always use Limit=1, or Limit=10, etc. Point is, we are looking for a second entry and there is none.